usage:
```
spglib provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(Spglib CONFIG REQUIRED)
  target_link_libraries(main PRIVATE Spglib::symspg Spglib::fortran)

spglib provides pkg-config modules:

  # The spglib library
  spglib
```

When testing usage, the following error occurs:
```
1> [CMake] CMake Error at test/CMakeLists.txt:7 (target_link_libraries):
1> [CMake]   Target "main" links to:
1> [CMake] 
1> [CMake]     Spglib::fortran
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
```

<details>
<summary><b>test.cpp</b></summary>
<pre>
#include &#60;iostream&#62;
#include "spglib.h"

using namespace std;

int main()
{
        cout << "Hello CMake." << endl;
        return 0;
}
</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

set(CMAKE_TOOLCHAIN_FILE "G:/spg/vcpkg/scripts/buildsystems/vcpkg.cmake")

project ("test")

add_library (main "test.cpp")

find_package(Spglib CONFIG REQUIRED)
target_link_libraries(main PRIVATE Spglib::symspg Spglib::fortran)
</pre></details>

--- a/ports/spglib/portfile.cmake
+++ b/ports/spglib/portfile.cmake
@@ -1,0 +1,26 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO spglib/spglib
+    REF "v${VERSION}"
+    SHA512 15c0ced6168a436468d1f9db28bb93f3ff130467cd1f0b966cb9731d36be3d9877b3452561dbace3242351b7c9b41d41930a76ca2278f00c1b45620c06ee93e0
+    HEAD_REF develop
+)
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" SPGLIB_SHARED_LIBS)
+
+vcpkg_cmake_configure(SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+    -DSPGLIB_WITH_TESTS=OFF
+    -DSPGLIB_SHARED_LIBS=${SPGLIB_SHARED_LIBS}
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/Spglib)
+vcpkg_fixup_pkgconfig()
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+
+# handle copyright
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/COPYING")

--- a/ports/spglib/usage
+++ b/ports/spglib/usage
@@ -1,0 +1,9 @@
+spglib provides CMake targets:
+
+    find_package(Spglib CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE Spglib::symspg)
+
+spglib provides pkg-config modules:
+
+    # The spglib library
+    spglib

--- a/ports/spglib/vcpkg.json
+++ b/ports/spglib/vcpkg.json
@@ -1,0 +1,17 @@
+{
+  "name": "spglib",
+  "version-semver": "2.4.0",
+  "description": "C library for finding and handling crystal symmetries",
+  "homepage": "https://spglib.readthedocs.io/en/latest/",
+  "license": "BSD-3-Clause",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ]
+}

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -8396,6 +8396,10 @@
       "baseline": "1.2.1",
       "port-version": 1
     },
+    "spglib": {
+      "baseline": "2.4.0",
+      "port-version": 0
+    },
     "spine-runtimes": {
       "baseline": "4.1.0",
       "port-version": 0

--- a/versions/s-/spglib.json
+++ b/versions/s-/spglib.json
@@ -1,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "72a361224c8e7cc3dd14810ae402a410aad4c0bb",
+      "version-semver": "2.4.0",
+      "port-version": 0
+    }
+  ]
+}

